pkg/apis/flow/v1alpha1/xmltojsontransformation_types.go

```diff
@@ -0,0 +1,105 @@
+// +genreconciler
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// XMLToJSONTransformation is the schema for the event transformer.
+type XMLToJSONTransformation struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec holds the desired state of the XMLToJSONTransformation (from the client).
+	// +optional
+	Spec XMLToJSONTransformationSpec `json:"spec,omitempty"`
+
+	// Status communicates the observed state of the XMLToJSONTransformation (from the controller).
+	// +optional
+	Status XMLToJSONTransformationStatus `json:"status,omitempty"`
+}
+
+var (
+	// Check that XMLToJSONTransformation can be validated and defaulted.
+	_ apis.Validatable   = (*XMLToJSONTransformation)(nil)
+	_ apis.Defaultable   = (*XMLToJSONTransformation)(nil)
+	_ kmeta.OwnerRefable = (*XMLToJSONTransformation)(nil)
+	// Check that the type conforms to the duck Knative Resource shape.
+	_ duckv1.KRShaped = (*XMLToJSONTransformation)(nil)
+)
+
+// XMLToJSONTransformationSpec holds the desired state of the XMLToJSONTransformation (from the client).
+type XMLToJSONTransformationSpec struct {
+	// EventOptions for targets
+	EventOptions *EventOptions `json:"eventOptions,omitempty"`
+
+	// Sink is a reference to an object that will resolve to a uri to use as the sink.
+	Sink duckv1.Destination `json:"sink,omitempty"`
+}
+
+// EventOptions modifies CloudEvents management at Targets.
+type EventOptions struct {
+	// PayloadPolicy indicates if replies from the target should include
+	// a payload if available. Possible values are:
+	//
+	// - always: will return a with the reply payload if avaliable.
+	// - errors: will only reply with payload in case of an error.
+	// - never: will not reply with payload.
+	//
+	// +optional
+	PayloadPolicy *cloudevents.PayloadPolicy `json:"payloadPolicy,omitempty"`
+}
+
+const (
+	// XMLToJSONTransformationConditionReady is set when the revision is starting to materialize
+	// runtime resources, and becomes true when those resources are ready.
+	XMLToJSONTransformationConditionReady = apis.ConditionReady
+)
+
+// XMLToJSONTransformationStatus communicates the observed state of the XMLToJSONTransformation (from the controller).
+type XMLToJSONTransformationStatus struct {
+	duckv1.SourceStatus `json:",inline"`
+
+	// Address holds the information needed to connect this Addressable up to receive events.
+	// +optional
+	Address *duckv1.Addressable `json:"address,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// XMLToJSONTransformationList is a list of XMLToJSONTransformation resources
+type XMLToJSONTransformationList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []XMLToJSONTransformation `json:"items"`
+}
+
+// GetStatus retrieves the status of the resource. Implements the KRShaped interface.
+func (t *XMLToJSONTransformation) GetStatus() *duckv1.Status {
```

Comment:
Interface implementations are typically inside `lifecycle.go`.
`types.go` is for types.